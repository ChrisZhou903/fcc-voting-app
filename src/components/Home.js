import React, { PureComponent } from 'react'
import { object } from 'prop-types'
import { Link } from 'react-router-dom'
import { Button } from 'antd'

import './Home.css'

class Home extends PureComponent {
  static propTypes = {
    user: object,
  }

  static defaultProps = {
    user: {},
  }

  render() {
    const { user } = this.props
    const name = (user && user.displayName) || ''
    return (
      <div className="App">
        <div className="App-header">
          <img src="/assets/images/logo.svg" className="App-logo" alt="logo" />
          <h2>
            Welcome {name} to FCC VOTING APP
          </h2>
        </div>
        <p className="App-intro">
          There is no votes yet.<br />
          <Link to="/vote/new">
            <Button type="primary" icon="file-add" className="add-vote">
              Add a new vote
            </Button>
          </Link>
        </p>
      </div>
    )
  }
}

export default Home
