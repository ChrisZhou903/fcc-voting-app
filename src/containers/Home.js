import React, { PureComponent } from 'react'
import { object } from 'prop-types'
import { connect } from 'react-redux'
import logo from '../logo.svg'
import './Home.css'

class Home extends PureComponent {
  static propTypes = {
    user: object, // eslint-disable-line
  }

  render() {
    const { user } = this.props
    const name = user.name || ''
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>Welcome { name } to React</h2>
        </div>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
      </div>
    )
  }
}

const mapStateToProps = state => ({
  user: state.user,
})

export default connect(mapStateToProps, {})(Home)
