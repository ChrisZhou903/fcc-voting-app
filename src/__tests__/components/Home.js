import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import Home from '../../components/Home'

const setup = (user = {}) => {
  const component = <Home user={user} />
  const wrapper = shallow(component)

  return {
    component,
    wrapper,
  }
}

describe('Home component', () => {
  it('should render without errors', () => {
    const { wrapper } = setup()

    expect(wrapper).toBeTruthy()
  })

  it('should render correct things', () => {
    const { wrapper } = setup()

    expect(wrapper.find('.App').length === 1).toBe(true)
  })

  it("should render right thing and don't change unexpected", () => {
    const { component } = setup()
    const tree = renderer.create(component).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
