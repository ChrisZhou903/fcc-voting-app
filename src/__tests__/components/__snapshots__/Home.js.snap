// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should render right thing and don't change unexpected 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="/assets/images/logo.svg"
    />
    <h2>
      Welcome 
      
       to FCC VOTING APP
    </h2>
  </div>
  <p
    className="App-intro"
  >
    There is no votes yet.
    <br />
    <a
      href="/vote/new"
      onClick={[Function]}
    >
      <button
        className="ant-btn add-vote ant-btn-primary"
        onClick={[Function]}
        onMouseUp={[Function]}
        type="button"
      >
        <i
          className="anticon anticon-file-add"
        />
        <span>
          Add a new vote
        </span>
      </button>
    </a>
  </p>
</div>
`;
